From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ralf Habacker <ralf.habacker@freenet.de>
Date: Wed, 28 May 2025 01:12:26 +0200
Subject: [PATCH] ws2_32: Add note in bind() for AF_UNIX sockets

Origin: https://gitlab.winehq.org/wine/wine/-/merge_requests/7650 (head 30718c3ce0c1)
Topic: afunix
---
 dlls/ws2_32/socket.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/ws2_32/socket.c b/dlls/ws2_32/socket.c
index 7a5720583d38d6e7ae3e80b8f9c9b9109a8009cc..818ad66e30a9717a5bf2dec3d8d694912f093b7e 100644
--- a/dlls/ws2_32/socket.c
+++ b/dlls/ws2_32/socket.c
@@ -1340,6 +1340,8 @@ int WINAPI bind( SOCKET s, const struct sockaddr *addr, int len )
 
     if (addr->sa_family == AF_UNIX && *addr->sa_data)
     {
+        /* The corresponding unix path is appended to a buffer with
+         * the structure sockaddr_un and can have a length of <= PATH_MAX */
         struct sockaddr_un sun = { 0 };
         WCHAR *sun_pathW;
         memcpy(&sun, addr, len);
